### ISSUE TITLE
[feature] provide msbuild property sheets to consumers

### ISSUE THREAD

**Issue by dvirtz** [conan-io/conan#12787]
### What is your suggestion?

I'm trying to convert a Nuget dependency to conan. Nuget packages can provide custom build targets or properties to consumers in the form of `<package_id>.targets` or `<package_id>.props` files (see https://learn.microsoft.com/en-us/nuget/concepts/msbuild-props-and-targets). 
I think conan should be able to provide such ability, e.g. by adding an `msbuild_build_modules`  property, similar to `cmake_build_modules`, and including those modules in `MSBuildDeps` generator outputs.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @dvirtz 

This seems like a possible feature, but it would be a bit difficult to prioritize, due to our lack of knowledge of these MSBuild features. You could try to help by providing some further details about how this feature would like:
- A simple ``.props`` or ``.targets`` (or both if they will be handled slightly different), with some testable minimal (like printing "hello world" functionality)
- What kind of property, value should it generate in the generated ``MSBuildDeps`` files
- How this target/property is used, executed in the consumer project.

Depending on the above, it might be possible that this feature is not even necessary if MSBuildDeps provide the path to the dependency as a property (it already does), so the consumer project can already access that file easily? Thanks!

**Comment by dvirtz** [comment]
A simple example can be found at https://github.com/fredemmott/cmake-cpp-winrt-winui3. Currently it uses Nuget dependencies: 
https://github.com/fredemmott/cmake-cpp-winrt-winui3/blob/a09d35b17b31ec07aba437a6346b6a6db6f2a91f/src/CMakeLists.txt#L26-L33
and I'm trying to convert it to use conan.

`cppwinrt`, for example, provides https://github.com/microsoft/cppwinrt/blob/0a6cb062e2151cf6c8f357aa8ef735e359f8a98c/nuget/Microsoft.Windows.CppWinRT.props and https://github.com/microsoft/cppwinrt/blob/0a6cb062e2151cf6c8f357aa8ef735e359f8a98c/nuget/Microsoft.Windows.CppWinRT.targets which, among other things, instruct MSBuild to use `cppwinrt` for compiling MIDL files.

I'm not an MSBuild expert either, but I think it's possible to simply add an [Import](https://learn.microsoft.com/en-us/visualstudio/msbuild/import-element-msbuild) element in the generated files pointing to those property sheets.

**Comment by memsharded** [comment]
> I'm not an MSBuild expert either, but I think it's possible to simply add an [Import](https://learn.microsoft.com/en-us/visualstudio/msbuild/import-element-msbuild) element in the generated files pointing to those property sheets.

yes, ``Import`` is doable. I was wondering about how this imported functionality is actually used in the consumer project. I'd need some real simple example of this that I could use to build a test. I have had a google search for example, but docs are not great, and I couldn't find something straightforward to use.

**Comment by memsharded** [comment]
Closing this as not planned at the moment, it doesn't seem to have enough demand from other users, and also we lack the expertise and resources to prioritize it.

If this is still desired or needed, we would need someone with expertise to push for it, we could help and guide with the integration, the tests, the style, etc.

Thanks for the feedback!
